### PR TITLE
Feature: Add new linkable component

### LINKED ISSUE #130: List all related components that can be link to a selected component
## Description

**As a** user, when I select a component

**I want** to have a way to list all related components that can be link to the selected

**so that** I select one of them to create a linked component to my selected one

### PR DESCRIPTION
**Pull request by Hundraw**
Close #130

### CODE REVIEW

File: src/draw/DefaultDrawer.js [hunk 1]
```diff
@@ -1273,7 +1283,116 @@ class DefaultDrawer {
...
+      .on('mouseleave', function onLeave() {
+        d3.select(this)
+          .select('rect')
+          .attr('fill', 'lightgrey')
+          .attr('cursor', 'default');
+      })
+      .on('click', (event, data) => {
+        this.actions.linkCreation.source = this.actions.selection.current;
+        const componentId = this.pluginData.addComponent(data);
+
```
Comment (Zorin95670) [inline src/draw/DefaultDrawer.js:1383]: remove empty line

File: src/draw/DefaultDrawer.js [hunk 2]
```diff
@@ -1273,7 +1283,116 @@ class DefaultDrawer {
...
+   *
+   * @param {ComponentDefinition[]} definitions - List of component definitions.
+   */
+  initializeCreateLinkableComponentMenu(definitions) {
+    d3.select('#linkable-menu')?.remove();
+
+    let maxTextWidth = 0;
+    const buttonPadding = 5;
+    const iconSize = 20;
+
```
Comment (Zorin95670) [inline src/draw/DefaultDrawer.js:1296]: remove empty line

File: src/draw/DefaultDrawer.js [hunk 3]
```diff
@@ -1273,7 +1283,116 @@ class DefaultDrawer {
...
+    const actionMenu = document.querySelector('#action-menu');
+
```
Comment (Zorin95670) [inline src/draw/DefaultDrawer.js:1298]: remove empty line

File: src/draw/DefaultDrawer.js [hunk 4]
```diff
@@ -1273,7 +1283,116 @@ class DefaultDrawer {
...
+      })
+      .on('mouseleave', function onLeave() {
+        d3.select(this)
+          .select('rect')
+          .attr('fill', 'lightgrey')
+          .attr('cursor', 'default');
+      })
+      .on('click', (event, data) => {
+        this.actions.linkCreation.source = this.actions.selection.current;
+        const componentId = this.pluginData.addComponent(data);
```
Comment (Zorin95670) [inline src/draw/DefaultDrawer.js:1382]: break line before

File: src/draw/DefaultDrawer.js [hunk 5]
```diff
@@ -1273,7 +1283,116 @@ class DefaultDrawer {
...
+
+        const component = this.pluginData.getComponentById(componentId);
+
+        component.path = this.actions.linkCreation.source.path;
+
+        this.draw(this.rootId);
+        this.actions.linkCreation.target = d3.select(`#${componentId}`).datum().data;
+        this.createLink();
```
Comment (Zorin95670) [inline src/draw/DefaultDrawer.js:1390]: verify if update event is sent

File: changelog.md [hunk 6]
```diff
@@ -5,6 +5,15 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html)
 
+
```
Comment (Zorin95670) [inline changelog.md:8]: too much empty line

**Review by Zorin95670 — CHANGES_REQUESTED** [review]

File: src/draw/DefaultDrawer.js (hunk 1 above)
Comment (Hundraw) [reply src/draw/DefaultDrawer.js:1383]: Done

File: src/draw/DefaultDrawer.js (hunk 2 above)
Comment (Hundraw) [reply src/draw/DefaultDrawer.js:1296]: Done

File: src/draw/DefaultDrawer.js (hunk 3 above)
Comment (Hundraw) [reply src/draw/DefaultDrawer.js:1298]: Done

File: src/draw/DefaultDrawer.js (hunk 4 above)
Comment (Hundraw) [reply src/draw/DefaultDrawer.js:1382]: Done

File: changelog.md (hunk 6 above)
Comment (Hundraw) [reply changelog.md:8]: Done

File: src/draw/DefaultDrawer.js (hunk 5 above)
Comment (Hundraw) [reply src/draw/DefaultDrawer.js:1390]: Done

**Review by Zorin95670 — APPROVED** [review]

**Pull request merged**